### PR TITLE
Finalize v1.2

### LINKED ISSUE #7: [EPIC] As a TA, I can manage my students in my contacts so I can keep in contact with them.
* As a TA, I can add students to my contacts.
* As a TA, I can remove students from my contacts.
* As a TA, I can view students in my contacts.

### LINKED ISSUE #8: [EPIC] As a TA, I can manage the modules I am teaching, so that I can monitor the modules I am teaching this semester.
* As a TA, I can add modules.
* As a TA, I can remove modules.
* As a TA, I can view modules.

### LINKED ISSUE #9: [EPIC] As a TA, I can manage the lessons I am teaching, so that I can monitor the lessons I have to teach.
* As a TA, I can add lessons.
* As a TA, I can remove lessons.
* As a TA, I can view lessons.

### LINKED ISSUE #92: Finalize v1.2
To check:
* can add all 3 entities
* can delete all 3 entities
* can list all 3 entities
* All 3 entities are stored in database

### PR DESCRIPTION
**Pull request by lkldev**
resolves #92 

closes #7 , closes #8, closes #9

Updated code to solve some bugs and output display. The main commits for this PR are:
* Fix bugs in delete and list where the entity is not parsed correctly
* Update command result formatting and standardize error messages
* Fix checks for module code and class group id to allow lowercases
* Update student isSameStudent method to check student ID instead
* Update parser to allow optional telegram
* Update UI to display telegram only if it is not empty
* Update UG to indicate that telegram is optional

### CODE REVIEW

**Comment by EvaderFati** [comment]
LGTM

**Review by EvaderFati — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -82,9 +86,11 @@ public static Name parseName(String name) throws ParseException {
      *
      * @throws ParseException if the given {@code phone} is invalid.
```
Comment (jxt00) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:87]: Should `phone` be `telegram` instead?

**Review by jxt00 — COMMENTED** [review]
Looks good!

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 1 above)
Comment (lkldev) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java]: Resolved in latest commit

**Review by jxt00 — APPROVED** [review]
LGTM 👍 Thanks!

**Pull request merged**